### PR TITLE
MRG: Release candidate 2.0.2

### LINKED ISSUE #373: Bug fix release
#370 shows there is at least one nibabel package maintainer supporting numpy 1.10 in the wild. Once #367 is in, I believe we'll have full Python 3.5 and numpy 1.10 support.

Does it make sense to have a bug fix release, or push on to 2.1 or 3.0?

### PR DESCRIPTION
**Pull request by effigies**
This is a cherry-picked branch off of maint/2.0.x of the bug-fix pull-requests since 2.0.1. The only PR that needed a little massaging was #363.

There are a number of little DOC and TST fixes that could probably also fit in, if people think it makes sense.

Closes #373

### CODE REVIEW

File: Changelog [hunk 1]
```diff
@@ -24,6 +24,23 @@ and Stephan Gerhard (SG).
 
 References like "pr/298" refer to github pull request numbers.
 
+* Upcoming
+
+  * Trackvis reader will now allow final streamline to have fewer points that
```
Comment (matthew-brett) [inline Changelog]: Trackvis changes not included in fact.

Comment (effigies) [inline Changelog]: Would you rather pull in the changes or drop the changelog entry?

Comment (matthew-brett) [inline Changelog]: Just drop the changelog entry I guess, it is a minor API change.

**Comment by effigies** [comment]
Removed the changelog entries.

@yarikoptic @ignatenkobrain Any extra commits in Debian/Fedora packages that you'd like to see in this release?

**Comment by yarikoptic** [comment]
On Fri, 06 Nov 2015, Chris Markiewicz wrote:

> Removed the changelog entries.

> [1]@yarikoptic [2]@ignatenkobrain Any extra commits in Debian/Fedora packages
> that you'd like to see in this release?

I don't think I have any patches BUT I would appreciate if you give me
one day -- I will test 2.0.1-148-gdd4e725  regarding different
debian/ubuntu releases and possibly breaking dependents

-- 
Yaroslav O. Halchenko
Center for Open Neuroscience     http://centerforopenneuroscience.org
Dartmouth College, 419 Moore Hall, Hinman Box 6207, Hanover, NH 03755
Phone: +1 (603) 646-9834                       Fax: +1 (603) 646-1419
WWW:   http://www.linkedin.com/in/yarik

**Comment by effigies** [comment]
Sounds good. I don't think we're in a rush, so just ping this thread when you're ready to go.

**Comment by ignatenkobrain** [comment]
I don't see new problems, but would be great if you will fix import for pydicom. Because we need to do import pydicom instead of import dicom.

**Comment by ignatenkobrain** [comment]
https://darcymason/pydicom/blob/master/setup.py#L11-L15

**Comment by effigies** [comment]
That means something like #351. I may have a few minutes to give that a try, but we'll need to test.

@matthew-brett Is it easy to make a new Travis test that would test against pydicom 1.0?

**Comment by yarikoptic** [comment]
FWIW -- everything looked splendid across debians for me with 2.0.1-148-gdd4e725

**Comment by effigies** [comment]
@yarikoptic Not sure quite how to interpret the string `2.0.1-148-gdd4e725`. I'm guessing that means applying this branch to your last debian release?

**Comment by yarikoptic** [comment]
On Fri, 06 Nov 2015, Chris Markiewicz wrote:

> [1]@yarikoptic Not sure quite how to interpret the string 2.0.1-148-gdd4e725.
> I'm guessing that means applying this branch to your last debian release?

    $> git show 2.0.1-148-gdd4e725 
    commit dd4e725156206661aa218db9b3be9c2f02246ecf
    Author: Matthew Brett <matthew.brett@gmail.com>
    Date:   Thu Nov 5 18:24:45 2015 -0800

        DOC: remove trailing sentence
        
        Fixes gh-374.

that was the output of git describe  on master at that point, e.g. as 

    $> git describe dd4e725156206661aa218db9b3be9c2f02246ecf
    2.0.1-148-gdd4e725

**Comment by effigies** [comment]
Oh, I see. Just FYI this PR is just cherry-picked bug fixes on top of the maint/2.0.x branch, not everything that's gone into master.

If people would rather just package master into a new release, that's fine, too. These were almost clean cherry-picks, so I've sunk very little work into this branch.

**Comment by ignatenkobrain** [comment]
@effigies can you also backport pydicom's patch to this PR? It's required oneline massaging only.

**Comment by matthew-brett** [comment]
Chris - any interest in being the release manager for this one?  Instructions here : https://github.com/nipy/nibabel/blob/master/doc/source/devel/make_release.rst - but I am sure they could be improved.

**Comment by effigies** [comment]
@matthew-brett Sure. Thanks for the notes.

@ignatenkobrain pydicom 1.0 is in.

**Comment by ignatenkobrain** [comment]
@effigies cool, thanks! ping me when you will release new version. I will immediately update one in Fedora.

**Comment by effigies** [comment]
@matthew-brett Running into this issue on `make sdist-tests`:

    ======================================================================
    ERROR: nibabel.tests.test_deprecated.test_futurewarning_mixin
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
        self.test(*self.arg)
      File "/tmp/tmp5DyxOV/pylib/nibabel/tests/test_deprecated.py", line 37, in test_futurewarning_mixin
        warn = warns.pop(0)
    IndexError: pop from empty list

    ======================================================================
    ERROR: nibabel.tests.test_minc1.test_old_namespace
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
        self.test(*self.arg)
      File "/tmp/tmp5DyxOV/pylib/nibabel/tests/test_minc1.py", line 110, in test_old_namespace
        assert_equal(warns.pop(0).category, FutureWarning)
    IndexError: pop from empty list

**Comment by matthew-brett** [comment]
Hum.  I can look at that tomorrow.  Maybe we can drop those tests soon
though.  Any interest in porting over the INSTALL_TYPE machinery from the
nipy travis setup?  That tests the different installation methods including
sdist...​

**Comment by effigies** [comment]
With #367 closed I'd say this is ready to move forward. (I don't see a short way forward with #362, so I'm inclined to leave that to another release.)

I'll be wrapping this up in the next couple of days, so I'd appreciate input from anybody who wants to see something from `master` in this release.

**Comment by bcipolli** [comment]
For #362 seems like an important case to me, as the HCP dataset is pretty popular. I agree if there's no simple way forward, we should just release without it.

I just finished my meeting in New Orleans, I can take a look at it this morning and see if I can set up a test that works for Travis and propose a workaround.

**Comment by effigies** [comment]
I guess another option is to pull in the 3.4 gzip code in `external`, which would be less dependent on a gzip executable.

**Comment by bcipolli** [comment]
https://bugs.python.org/issue25626

**Comment by matthew-brett** [comment]
Great - thanks a lot for doing that.​

**Comment by effigies** [comment]
@bcipolli @surchs Would you mind testing this branch with the `NIPY_EXTRA_TESTS=slow` environment variable set, as well? I added a commit that's just #383 squashed and backported.

**Comment by bcipolli** [comment]
Worked like a champ!

**Comment by effigies** [comment]
Cool. Thanks for the validation.

**Comment by effigies** [comment]
Okay. I think code-wise, this is ready to go. @matthew-brett could you walk me through the build-bots? I'm not quite sure how to go about this. (Also `ssh buildbot@nipy.bic.berkeley.edu echo` doesn't work, so I don't think my key is in yet.)

And should I `merge --ff` into `maint/2.0.x` or use this PR? Any preference?

**Comment by matthew-brett** [comment]
Chris - I updated the instructions here : https://github.com/nipy/nibabel/pull/386 - does that doc make sense to you?

**Comment by matthew-brett** [comment]
For merging, my preference is always to do non-ff merge unless there's a good reason - to keep the set of commits as an identifiable set of work.

**Comment by effigies** [comment]
Those instructions look good. I've been able to force builds (on the buildbot machine, not via the web interface), but haven't figured out pushing the `NIPY_EXTRA_TESTS` environment variable.

**Comment by matthew-brett** [comment]
Did you manage to get the web interface working in the end?   It's less useful as you have to push something to the main repo before you can build it - for example you can build a tag once it's pushed, but you usually want to try building stuff before it's pushed.

There's no easy way to push an environment variable, but in any case, the smaller Windows machine probably can't run the big test - the larger 64-bit machine probably can.   I'll try enabling that by default on the larger Windows machine.

**Comment by matthew-brett** [comment]
See for example env vars defined here : http://nipy.bic.berkeley.edu/builders/nibabel-bdist64-27/builds/43/steps/shell/logs/stdio

**Comment by matthew-brett** [comment]
I had to do this by logging into the machine remotely and setting the enviroment variable in the users profile.   Can give you access to this machine if you need it.

**Comment by matthew-brett** [comment]
Oops - even for the 6GB windows machine it was too much:

http://nipy.bic.berkeley.edu/builders/nibabel-bdist64-27/builds/43/steps/shell_6/logs/stdio

```
======================================================================
ERROR: nibabel.tests.test_nifti1.test_large_nifti1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\buildslaves\mike-win7-64\nibabel-bdist64-27\build\venv\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\Python27\lib\site-packages\numpy\testing\decorators.py", line 146, in skipper_func
    return f(*args, **kwargs)
  File "c:\buildslaves\mike-win7-64\nibabel-bdist64-27\build\venv\lib\site-packages\nibabel-2.1.0.dev0-py2.7-win-amd64.egg\nibabel\tests\test_nifti1.py", line 1257, in test_large_nifti1
    data = load('test.nii.gz').get_data()
  File "c:\buildslaves\mike-win7-64\nibabel-bdist64-27\build\venv\lib\site-packages\nibabel-2.1.0.dev0-py2.7-win-amd64.egg\nibabel\spatialimages.py", line 572, in get_data
    data = np.asanyarray(self._dataobj)
  File "c:\Python27\lib\site-packages\numpy\core\numeric.py", line 287, in asanyarray
    return array(a, dtype, copy=False, order=order, subok=True)
  File "c:\buildslaves\mike-win7-64\nibabel-bdist64-27\build\venv\lib\site-packages\nibabel-2.1.0.dev0-py2.7-win-amd64.egg\nibabel\arrayproxy.py", line 145, in __array__
    raw_data = self.get_unscaled()
  File "c:\buildslaves\mike-win7-64\nibabel-bdist64-27\build\venv\lib\site-packages\nibabel-2.1.0.dev0-py2.7-win-amd64.egg\nibabel\arrayproxy.py", line 140, in get_unscaled
    mmap=self._mmap)
  File "c:\buildslaves\mike-win7-64\nibabel-bdist64-27\build\venv\lib\site-packages\nibabel-2.1.0.dev0-py2.7-win-amd64.egg\nibabel\volumeutils.py", line 518, in array_from_file
    n_read = infile.readinto(data_bytes)
  File "c:\Python27\Lib\gzip.py", line 256, in read
    self._read(readsize)
  File "c:\Python27\Lib\gzip.py", line 308, in _read
    self._add_read_data( uncompress )
  File "c:\Python27\Lib\gzip.py", line 326, in _add_read_data
    self.extrabuf = self.extrabuf[offset:] + data
MemoryError
```

**Comment by effigies** [comment]
Yeah. I can run the test on a 24GB machine but not an 8GB machine. Maybe if we just have one of the build slaves set to always run with `NIPY_EXTRA_TESTS=slow,bigmem`, and we make sure to run against that one before releases, that will be enough?

Also, I set up a pypi user `cjmarkie`, in case you need to add upload permissions.

**Comment by matthew-brett** [comment]
OK - done for pypi.

I enabled slow,bigmem test for the big Mac buildslave:

https://github.com/nipy/nibotmi/commit/2322a7679882660479845cba22d5e3b271e7c5d5

(I had to log on and update the relevant file on that machine, then restart the buildbot slave).

I updated the release instructions adding the suggestion to run on this machine first:

https://github.com/nipy/nibabel/pull/386/files#diff-d13ddc96596291d914cd84c6165cc10bR130

**Comment by effigies** [comment]
Great. [Build successful](https://nipy.bic.berkeley.edu/builders/nibabel-py2.7-osx-10.10/builds/29).

I think I should be good to go. I'll let you know if I run into any more issues.

**Comment by effigies** [comment]
@matthew-brett Release pushed. Hopefully I didn't somehow miss anything too egregious. Will make notes on #386.

@ignatenkobrain [2.0.2 is released](https://github.com/nipy/nibabel/releases/tag/2.0.2). Source is also on [pypi](https://pypi.python.org/pypi/nibabel)

**Pull request merged**